### ISSUE TITLE
quite high cpu usage compared to other docker containers

### ISSUE THREAD

**Issue by rodeho** [louislam/uptime-kuma#2710]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

First of all, what a great project, really loving it!

My kuma is running fine, despite the fact that its using quite a lot CPU for the tasks. Is there any way to drill down the information to the source? 

Its almost of the time at +/- 3% with ups and downs.  Smokeping got arround 100+ hosts in the config, and kuma only 36. I already migrated them to ICMP only in favor to reduce the CPU load.
```

NAME               CONTAINER      CPU %     MEM USAGE / LIMIT
smokeping          5aa205174233   0.32%     128.2MiB / 3.838GiB
librenms           51c12f074f96   0.01%     155.1MiB / 3.838GiB
unifi-controller   d97de21f888c   0.56%     1.23GiB / 3.838GiB
uptime-kuma        78a94a210c00   3.25%     201.4MiB / 3.838GiB
mariadb            611bd19b14bd   0.02%     93.51MiB / 3.838GiB
iperf              d948d9a42ffd   0.00%     916KiB / 3.838GiB
speedtest          169cef138624   0.00%     69.89MiB / 3.838GiB
apache2            6e5d25c53a5f   0.01%     8.258MiB / 3.838GiB
watchtower         c099a80e3b10   0.00%     8.953MiB / 3.838GiB
pagermon           ef7b1dcb5932   0.22%     136.9MiB / 3.838GiB
```

When looking at htop in the container the server.js sometimes peaks to 100%. 

![image](https://user-images.githubusercontent.com/16855935/216006202-87ab10d5-6dab-4f37-836e-35ca03affbbc.png)

### 👟 Reproduction steps

Just start kuma-uptime as a container.

### 👀 Expected behavior

Lower CPU load, somewhere there is some taks within kuma thats eating more power than it should for such small installation.

Please lets dig into this and try to find the root cause, could you help me?

### 😓 Actual Behavior

Quite high 

### 🐻 Uptime-Kuma Version

1.19.6

### 💻 Operating System and Arch

VM with Ubuntu -> with Docker-CE

### 🌐 Browser

any, chrome or Safari

### 🐋 Docker Version

 Docker version 20.10.23, build 7155243

### 🟩 NodeJS Version

don't think I use this?

**Comment by lightingman117** [comment]
Maybe we can get some more data on this?

But I have 26 monitors (half http, half ping) running on Debian 10 on 2 cores from a TR3970X in XCP-NG
Heimdall is the only other container running.

CPU Utilization (AVG-2c) Peak: 13%, average(optically averaged, 1-hr period): 2.5%
366M

**Comment by lightingman117** [comment]
Again, Prove it.

I am up to 38 monitors (once group functionality comes, I'll do more).

3.5% AVG usage as reported by host (XCP-NG).

60sec intervals

I have 8 monitors on my 5900X UnRaid system, docker reports less than 0.5% usage.

**Comment by lightingman117** [comment]
Prove that your instance is using "unacceptable CPU usage"

You need to provide more data.

How much CPU utilization does your instance use right now?
What CPU?
What host?
What Version?

I have clean & old (updated) installs. All are functioning within my own expectations.
What are your expectations?

**Comment by lightingman117** [comment]
Deleted all monitors.
Back to 0.07% usage

![image](https://user-images.githubusercontent.com/54555519/232099926-eb38d8c7-9934-454d-924f-5014e93c8c3e.png)

Proper bug reports include details to help determine the issue. If developers can't recreate the issue how can they fix it?

**Comment by rodeho** [comment]
> Your percentages of around 1 percent didn't sound like much, until I realised that we're talking about a super high-end 16-core cpu here.
> 
> So I don't understand what I need to say to get you to understand: absolute percentages are meaningless. On a Raspberry Pi 4 or Celeron they will be ten times higher. The only thing you proved is that on a very expensive CPU you get lower numbers, great job. But that doesn't say anything about how well-optimized this software is.


Exactly! Running overhere on a small i3 laptop core (NUC).

**Comment by baderk** [comment]
I recently noticed higher than usual CPU from node server.js process and took me some time to figure out which container it was from and it was from this one. I don't think I've seen it this up before, but lately ( I want to say since the past month or so?) I've had it a bit higher. Not crazy high, but higher than usual. I didn't really add anything new, it's been a while since I added a new service to monitor.

The CPU usage is constantly above 50%, which as others have mentioned, a lot more than many other docker containers I have running. 

![Screenshot 2023-07-06 at 10 29 57 PM](https://github.com/louislam/uptime-kuma/assets/10656279/d50b542e-08bf-47e8-8f2b-77486eb7c120)

It's always hovering between 50-60%. Is there anything I can help with regarding investigating this?

I have i5-4690K CPU. Again, nothing crazy on usage on this CPU, but maybe a bit odd that it's using that much.
